lnwallet/aux_signer.go

```diff
@@ -0,0 +1,146 @@
+
+	// HtlcLeaf is the aux tap leaf that corresponds to the HTLC being
+	// signed/verified.
+	HtlcLeaf input.AuxTapLeaf
+}
+
+// AuxSigJob is a struct that contains all the information needed to sign an
+// HTLC for custom channels.
+type AuxSigJob struct {
+	// SignDesc is the sign desc for this HTLC.
+	SignDesc input.SignDescriptor
+
+	BaseAuxJob
+
+	// Resp is a channel that will be used to send the result of the sign
+	// job.
+	Resp chan AuxSigJobResp
+
+	// Cancel is a channel that should be closed if the caller wishes to
+	// abandon all pending sign jobs part of a single batch.
+	Cancel chan struct{}
+}
+
+// NewAuxSigJob creates a new AuxSigJob.
+func NewAuxSigJob(sigJob SignJob, keyRing CommitmentKeyRing, incoming bool,
+	htlc PaymentDescriptor, commitBlob fn.Option[tlv.Blob],
+	htlcLeaf input.AuxTapLeaf, cancelChan chan struct{}) AuxSigJob {
+
+	return AuxSigJob{
+		SignDesc: sigJob.SignDesc,
+		BaseAuxJob: BaseAuxJob{
+			OutputIndex: sigJob.OutputIndex,
+			KeyRing:     keyRing,
+			HTLC:        htlc,
+			Incoming:    incoming,
+			CommitBlob:  commitBlob,
+			HtlcLeaf:    htlcLeaf,
+		},
+		Resp:   make(chan AuxSigJobResp, 1),
+		Cancel: cancelChan,
+	}
+}
+
+// AuxSigJobResp is a struct that contains the result of a sign job.
+type AuxSigJobResp struct {
+	// SigBlob is the signature blob that was generated for the HTLC. This
+	// is an opaque TLV field that may contain the signature and other data.
+	SigBlob fn.Option[tlv.Blob]
+
+	// HtlcIndex is the index of the HTLC that was signed.
+	HtlcIndex uint64
+
+	// Err is the error that occurred when executing the specified
+	// signature job. In the case that no error occurred, this value will
+	// be nil.
+	Err error
+}
+
+// AuxVerifyJob is a struct that contains all the information needed to verify
+// an HTLC for custom channels.
+type AuxVerifyJob struct {
+	// SigBlob is the signature blob that was generated for the HTLC. This
+	// is an opaque TLV field that may contain the signature and other data.
+	SigBlob fn.Option[tlv.Blob]
+
+	BaseAuxJob
+
+	// Cancel is a channel that should be closed if the caller wishes to
+	// abandon the job.
+	Cancel chan struct{}
+
+	// ErrResp is a channel that will be used to send the result of the
+	// verify job.
+	ErrResp chan error
+}
+
+// NewAuxVerifyJob creates a new AuxVerifyJob.
+func NewAuxVerifyJob(sig fn.Option[tlv.Blob], keyRing CommitmentKeyRing,
+	incoming bool, htlc PaymentDescriptor, commitBlob fn.Option[tlv.Blob],
+	htlcLeaf input.AuxTapLeaf) AuxVerifyJob {
+
+	return AuxVerifyJob{
+		SigBlob: sig,
+		BaseAuxJob: BaseAuxJob{
+			KeyRing:    keyRing,
+			HTLC:       htlc,
+			Incoming:   incoming,
+			CommitBlob: commitBlob,
+			HtlcLeaf:   htlcLeaf,
+		},
+	}
+}
+
+// AuxSigner is an interface that is used to sign and verify HTLCs for custom
+// channels. It is similar to the existing SigPool, but uses opaque blobs to
+// shuffle around signature information and other metadata.
+type AuxSigner interface {
```

Comment:
would be nice to mock this and have some minimal test coverage

Reply:
Added this to the [tracking issue](https://github.com/lightningnetwork/lnd/issues/8769), since we currently don't have any coverage for those aux interfaces.